File: docs/DSA/neetcode150/src/go/daily_temperatures.go
package main

func dailyTemperatures(temperatures []int) []int {
	result := make([]int, len(temperatures))
	var stk []int
	var l int
	for r, temp := range temperatures {
		for len(stk) != 0 && temperatures[stk[len(stk)-1]] < temp {
			l = stk[len(stk)-1]
			stk = stk[:len(stk)-1]
			result[l] = r - l
		}
		stk = append(stk, r)
	}
	return result
}

File: docs/DSA/src/py/daily_temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: list[int]) -> list[int]:
        result = [0]*len(temperatures)
        stk = []
        for r, temp in enumerate(temperatures):
            while len(stk) != 0 and temperatures[stk[-1]] < temp:
                l = stk.pop()
                result[l] = r-l

            stk.append(r)
        return result